### ISSUE TITLE
Tool info module for validators

### ISSUE THREAD

**Issue by hernanponcedeleon** [sosy-lab/benchexec#746]
I'm preparing a PR to add dartagnan as a (violation) validator and a I have some doubts about how to retrieve the witness path in the python code.

While I can use `task.property_file` to retrieve the path of the property file, I'm not sure how to do the same for the path of the witness. I suspect somehow this information is given by `input_files`, but than can also be used for the path of the program so I'm not sure how to do this properly.

I checked the modules of other validators but I could not find any lead.

Also, AFAIK, there is no naming convention for validators. I might follow the `witness2test` naming because we are basically converting a violation witness into an execution (i.e. test) and encoding the execution into SMT to reduce the search space of our verification encoding.

**Comment by PhilippWendler** [comment]
BenchExec does not know anything about witnesses, so these do not exist in its data model.

`input_files` is just what the user configures as `input_files` in their task definitions. For software verification, this would typically be the program sources and not the witness (but one could do this in principle from the point of view of BenchExec, it would just be inconvenient probably).

For the existing validators, paths to witnesses are just configured as command-line arguments ([example](https://gitlab.com/sosy-lab/sv-comp/bench-defs/-/blob/a9bfd363f1c222165e09708585c5fdcfa1d47495/benchmark-defs/cpa-seq-validate-correctness-witnesses.xml#L16)). This would also be my recommendation.

We do not have any strict naming convention for tool-info modules in BenchExec, so just pick one that suits you, is intuitive for users, and not too general (to avoid collisions).

**Comment by hernanponcedeleon** [comment]
Ohhh then I guess this is much easier than I thought: I don't really need to add a new tool module but just create a new `benchmark-defs` using `dartagnan` as a tool and setting the corresponding option `-witness` with the value `../../results-verified/LOGDIR/${rundefinition_name}.${taskdef_name}/witness.graphml` (I guess this is fixed and depends on the configuration of the server where the competition is run), right?

**Comment by PhilippWendler** [comment]
Yes. The path just comes from how the competition organizer has his local setup, actually. This is basically defined and documented in [this repository](https://gitlab.com/sosy-lab/sv-comp/bench-defs), but not necessary to worry about.
